hypervisor/scripts/genconf.sh

```diff
@@ -26,6 +28,36 @@ if [ $? -ne 0 ]; then
     exit $?
 fi
 
+if ! which xsltproc ; then
+    echo "xsltproc cannot be found, please install it and make sure it is in your PATH."
+    exit 1
+fi
+
+transform() {
+    echo "Generating ${1}:"
+    xsltproc -o ${out}/scenarios/${scenario}/${1} --xinclude --xincludestyle ${tool_dir}/xforms/${1}.xsl ${unified_xml}
+    if [ $? -ne 0 ]; then
+        echo "Failed to generate ${1} with xsltproc!"
+        exit 1
+    fi
+    echo "${1} is generated using xsltproc succesfully!"
+}
+
+transform vm_configurations.c
+transform vm_configurations.h
+transform pt_intx.c
+transform ivshmem_cfg.h
+transform misc_cfg.h
+
+if which clang-format ; then
```

Comment:
@dbkinder clang-format is optional, it would compile without it. The script wouldn't exit.

We remove the indents formatting in xslt xforms. It put too much efforts to align the code with many hex characters "&#x9;" which represents tab in xml. In order to help developers or user to focus on embedded essential data to output files, the indents are no longer taking care by script itself.

@junjiemao1 @NanlinXie can discuss how to improve the document and let user/developer being aware of it.

Reply:
If clang-format is needed we can add it to the documentation as a required package.  That's a better DX experience instead of writing out the message saying clang-format is missing and please install it.

Reply:
Do we have optional package section? It's fine with me to make it required, it wouldn't change the machine code.
Just wondering shall we be more precise since it does not lead to compilation failure.

Reply:
I would agree with @dbkinder , let's add it to the list of packages to be installed. There is virtually no benefit in not installing it, just potential problems. With that in mind, I think it's better to make people install it by default although it is not strictly needed.

Reply:
If ``clang-format`` is really optional, then please remove the messages in the script saying clang-format wasn't run and you should install it.

Reply:
@dbkinder  The message did address that with if:
"Please install clang-format and format the generated files if those need to be included and reviewed."
Would this be sufficient to explain the reason of clang-format installation?

I would keep this explanation in case people really didn't install the clang-format.

@NanlinXie  could you help to address @gvancuts 's first comment below?

> @NanlinXie is this a new project/contribution guideline? I don't see `clang-format` as a tool we've asked developers to install (and use) as an ACRN pre-requisite, and it's not installed by packages on this list either: https://projectacrn.github.io/latest/getting-started/building-from-source.html#install-build-tools-and-dependencies. If it's required now, it should be added to the dependencies list and we should add this as a requirement in our contribution guidelines: https://projectacrn.github.io/latest/developer-guides/contribute_guidelines.html#coding-style.

Reply:
I just added clang-format to the list of packages to install in PR #5986